yt_dlp/extractor/rokfin.py

```diff
@@ -160,6 +169,157 @@ def _get_comments(self, video_id):
             if not raw_comments.get('content') or is_last or (page_n > pages_total if pages_total else is_last is not False):
                 return
 
+    def _perform_login(self, username, password):
+        LOGIN_PAGE_URL = 'https://secure.rokfin.com/auth/realms/rokfin-web/protocol/openid-connect/auth?client_id=web&redirect_uri=https%3A%2F%2Frokfin.com%2Ffeed&response_mode=fragment&response_type=code&scope=openid'
+        AUTHENTICATION_URL_REGEX_STEP_1 = r'\<form\s+[^>]+action\s*=\s*"(?P<authentication_point_url>https://secure\.rokfin\.com/auth/realms/rokfin-web/login-actions/authenticate\?[^"]+)"[^>]*>'
+
+        # In OpenID terms (https://openid.net/specs/), this program acts as a (public) Client (a.k.a. Client Application),
+        # the User Agent, and the Relying Party (RP) simultaneously, with the Identity Provider (IDP) being secure.rokfin.com.
+        # Rokfin uses KeyCloak (https://www.keycloak.org) as the OpenID implementation of choice.
+        #
+        # ---------------------------- CODE FLOW AUTHORIZATION ----------------------------
+        # https://openid.net/specs/openid-connect-core-1_0.html#CodeFlowAuth (Sec. 3.1.1)
+        #
+        # Authentication phase:
+        #
+        # Step 1: preparation
+        login_page = self._download_webpage(LOGIN_PAGE_URL, None, note='loading login page', fatal=False)
+        if not login_page:
+            return
+        authentication_point_url = unescapeHTML(re.search(AUTHENTICATION_URL_REGEX_STEP_1, login_page).group('authentication_point_url'))
+        if authentication_point_url is None:
+            self.report_warning('login failed unexpectedly: Rokfin extractor must be updated')
+            self._clear_cookies()
+            return
+
+        # Step 2 & 3: authentication
+        resp_body = self._download_webpage(
+            authentication_point_url, None, note='logging in', fatal=False, expected_status=404, encoding='utf-8',
+            data=urlencode_postdata({'username': username, 'password': password, 'rememberMe': 'off', 'credentialId': ''}))
+        # rememberMe=off resets the session when yt-dlp exits:
+        # https://web.archive.org/web/20220218003425/https://wjw465150.gitbooks.io/keycloak-documentation/content/server_admin/topics/login-settings/remember-me.html
```

Comment:
yt-dlp is not a browser!

Reply:
> rememberMe=off resets the session when yt-dlp exits:

This comment is wrong (since yt-dlp is not a browser). You can leave rememberme in the code, but delete the comment